usecases/decision_usecase.go

```diff
@@ -263,13 +269,39 @@ func (usecase *DecisionUsecase) CreateDecision(
 				fmt.Errorf("error storing decision: %w", err)
 		}
 
-		err := usecase.decisionWorkflows.AutomaticDecisionToCase(ctx, tx, scenario, decision)
+		webhookEventId := uuid.NewString()
+		err := usecase.webhookEventsSender.CreateWebhookEvent(ctx, tx, models.WebhookEventCreate{
+			Id:             webhookEventId,
+			OrganizationId: decision.OrganizationId,
+			EventContent:   models.NewWebhookEventDecisionCreated(decision.DecisionId),
+		})
 		if err != nil {
 			return models.DecisionWithRuleExecutions{}, err
 		}
+		sendWebhookEventId = append(sendWebhookEventId, webhookEventId)
+
+		caseWebhookEventId := uuid.NewString()
+		webhookEventCreated, err = usecase.decisionWorkflows.AutomaticDecisionToCase(
+			ctx,
+			tx,
+			scenario,
+			decision,
+			caseWebhookEventId)
+		if err != nil {
+			return models.DecisionWithRuleExecutions{}, err
+		}
+		if webhookEventCreated {
+			sendWebhookEventId = append(sendWebhookEventId, caseWebhookEventId)
+		}
 
 		return usecase.decisionRepository.DecisionWithRuleExecutionsById(ctx, tx, decision.DecisionId)
 	})
+
+	for _, webhookEventId := range sendWebhookEventId {
+		usecase.webhookEventsSender.SendWebhookEventAsync(ctx, webhookEventId)
+	}
+
+	return newDecision, err
```

Comment:
Here you do not take into account possible transaction error, is it intentionnal ?

Reply:
you are right, thanks for pointing it out